### PR TITLE
Move Remaining Styleguidist Stories to Storybook

### PR DESCRIPTION
**Pull request by SudharakaP**
Resolve https://github.com/opencollective/opencollective/issues/4313

### CODE REVIEW

**Review by Betree — COMMENTED** [review]
I have a crash on https://oc-storybook-demo-pr-6585.onrender.com/?path=/story/contribution-flow-steppayment--default-story, otherwise what I tested looks good. Even if not perfect, with this migration our current Storybook is a nice first version that we should try to merge as soon as we can to finally get rid of styleguidist. Next steps after this:
- Redirect styleguide.opencollective.com
- Get rid of everything related to styleguidist in the code

**Comment by SudharakaP** [comment]
> I have a crash on https://oc-storybook-demo-pr-6585.onrender.com/?path=/story/contribution-flow-steppayment--default-story, otherwise what I tested looks good. Even if not perfect, with this migration our current Storybook is a nice first version that we should try to merge as soon as we can to finally get rid of styleguidist. Next steps after this:
> 
>     * Redirect styleguide.opencollective.com
> 
>     * Get rid of everything related to styleguidist in the code

@Betree : I've just corrected the crash on that page. It seems that the `onChange` method handler was missing. Anyways, let me know if you see anything else or we could progress with the migration. 😄 

Let me investigate any missing pieces of styleguidiest to try to get rid of anything remaining. 😄

**Comment by SudharakaP** [comment]
> Get rid of everything related to styleguidist in the code

@Betree : I've removed all references to Styleguidist and added some documentation on how to add a new storybook component as well. Hope it looks good now. 😄

File: docs/styleguide.md [hunk 1]
```diff
@@ -10,16 +10,62 @@ npm run styleguide:dev
 
 ## Create a new component:
 
-Only components with a matching example markdown file in the `styleguide/examples/` directory will appear in the styleguide. After creating a new component in the `components/` directory (i.e. `components/NewComponent.js`), make an example markdown file to go with it (i.e. `styleguide/examples/NewComponent.md`).
+Storybook files are defined in `mdx` format and placed in the `stories/` folder. Adding a new story could simply be done by
```
Comment (Betree) [inline docs/styleguide.md:13]: Great job updating this documentation

**Review by Betree — APPROVED** [review]

File: docs/styleguide.md (hunk 1 above)
Comment (SudharakaP) [reply docs/styleguide.md:13]: Thanks, appreciated. 🤗 😄

**Pull request merged**